.autobahn.sh

```diff
@@ -0,0 +1,101 @@
+# A script that runs swift test followed by the autobahn test suite.
+
+# Generate the fuzzingclient.json file for the given tests
+fuzzing_client() {
+    TESTS=$1
+    echo "{\"outdir\": \"./reports/servers\", \"servers\": [{ \"url\": \"ws://127.0.0.1:9001\" }], \"cases\": [$TESTS],\"exclude-cases\": []}" > fuzzingclient.json
+}
+
+# Launch the WebSocket service and run a subset of Autobahn tests. On failures or unclean closures, exit with a non-zero code.
+run_autobahn()
+{
+    # The first argument holds the lists of tests to be run
+    TESTS=$1
+    NTESTS=$2
+
+    if [ `uname` == "Linux" ]; then
+        PLATFORM_SUBDIR="x86_64-unknown-linux"
+    else
+        PLATFORM_SUBDIR="x86_64-apple-macosx"
+    fi
+
+    # Launch the TestWebSocketService, save its pid
+    ./.build/$PLATFORM_SUBDIR/release/TestWebSocketService &
+    PID=$!
+
+    # Make sure the server has enough time to be up and running
+    sleep 5
+
+    # Generate the fuzzingclient.json
+    fuzzing_client $TESTS
+
+    # Run the autobahn fuzzingclient
+    wstest -m fuzzingclient
+
+    # Count the number of failed tests or unclean connection closures
+    FAILED_OR_UNCLEAN=`grep behavior reports/servers/index.json | cut -d':' -f2 | cut -d'"' -f2 | sort -u | xargs | grep -E "FAILED|UNCLEAN" | wc -l`
+    if [ $FAILED_OR_UNCLEAN -ne "0" ]; then
+        return $FAILED_OR_UNCLEAN
+    fi
+
+    TOTAL_TESTS=`grep behaviorClose reports/servers/index.json | wc -l`
+
+    echo "Executed $TOTAL_TESTS out of $NTESTS tests"
+    # Check if all tests completed
+    if [ $TOTAL_TESTS -ne $NTESTS ]; then
+        echo "All of the configured tests were not executed, possibly because the server wasn't available."
+        exit 1
+    fi
+
+    # Kill the service
+    kill $PID
+
+    # Remove the reports and the generated json file
+    rm -rf reports fuzzingclient.json
+}
+
+install_autobahn() {
+    if [ `uname` == "Linux" ]; then
+        apt-get update \
+            && apt-get -y upgrade \
+            && apt-get -y install sudo \
+            && sudo apt-get -y install python-pip \
+            && pip install autobahntestsuite
+    else
+        pip install autobahntestsuite
+    fi
+}
+
+# Run swift test
+travis_start "swift_test"
+swift test
+SWIFT_TEST_STATUS=$?
+travis_end
+if [ $SWIFT_TEST_STATUS -ne 0 ]; then
+    return $SWIFT_TEST_STATUS
+fi
+
+# Build TestWebSocketService
+echo "Building in release mode for autobahn testing"
+swift build -c release
```

Comment:
I think this part should also be wrapped in a travis_start / travis_end, as it's quite a sizeable chunk of output (particularly with BoringSSL).